native/cocos/renderer/pipeline/custom/LayoutGraphFwd.h

```diff
@@ -55,20 +55,28 @@ using UniformID = uint32_t;
 
 struct UniformData;
 struct UniformBlockData;
-
-using DescriptorID = uint32_t;
-
+struct NameLocalID;
 struct DescriptorData;
 struct DescriptorBlockData;
-struct DescriptorTableData;
+struct DescriptorSetLayoutData;
 struct DescriptorSetData;
 struct PipelineLayoutData;
 struct ShaderProgramData;
+struct RenderStageData;
 struct RenderPhaseData;
 struct LayoutGraphData;
 
 } // namespace render
 
 } // namespace cc
 
+namespace std {
+
+template <>
+struct hash<cc::render::NameLocalID> {
+    size_t operator()(const cc::render::NameLocalID& v) const noexcept;
```

Comment:
Will the hash value be bound to JS? If so, then can not use size_t.

Reply:
Implementation of hash is 32bit。Only the return type is size_t, which is consistent with std::hash